### ISSUE TITLE
BUG/FREQ-Shebang comment in .bashrc causes Fig to declare broken config.

### ISSUE THREAD

**Issue by gwiede** [withfig/fig#2195]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and replicated the issue again

### Operating system

macOS 12.6.2 (21G320)

### Expected behaviour

Lines of comments or whitespace between start of .bashrc file and first fig-added line should be ignored.  There are legitimate reasons for adding shebangs and other header comments ahead of fig's integration lines, fig should only care it has the first active​ line in the file.

### Actual behaviour

Presence of shebang comment line ahead of fig-added integration line triggers fig to complain integration broken.

### Steps to reproduce

 Add line consisting solely of...

#!/bin/env bash

...ahead of first fig-added integration line.

Note upon reboot (possibly even sooner) fig will report configuration problem and offer to fix.

REQUEST:

As stated above, fig should ignore all whitespace and comment lines between its first added integration line, and the beginning of the .bashrc (or whichever) script file.  Fig should not care that there is non-active comments and/or whitespace ahead of its additions.

Also, if there is a way to disable fig from specifically checking its shell integration was the first line, I couldn't find the way to disable that check.  I do think having the ability to disable that check (indepedent of the change requested) would also be useful, for cases where the user/admin KNOWS what they're adding must come ahead of any fig integration lines.

Thanks!

-John Wiederhirn
jfw@kompositor.com

### Environment

```yaml
fig-details:
  - 2.10.1-beta.0
hardware-info:
  - model:
  - model-id:
  - chip-id: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  - cores: 8
  - mem: 32.00 GB
os-info:
  - macOS 12.6.2 (21G320)
environment:
  - shell: /bin/bash
  - terminal: iterm
  - cwd: /Users/wiederhirn/gRepos/a10a20_1
  - exe-path: /Users/wiederhirn/.local/bin/fig
  - install-method: brew
  - env-vars:
    - FIG_PID: 74074
    - TERM: xterm-256color
    - SHELL: /bin/bash
    - FIGTERM_SESSION_ID: 705acd62-ec10-422a-8aa4-be762c888fb3
    - FIG_SET_PARENT_CHECK: 1
    - PATH: /Users/wiederhirn/gRepos/a10a20_1/pigweed/out/host/host_tools:/Users/wiederhirn/gRepos/a10a20_1/.environment/pigweed-venv/bin:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/luci:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/bazel:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/python/bin:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/python:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/arm/bin:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/arm:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/pigweed/bin:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/pigweed:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd/packages/a10a20:/Users/wiederhirn/gRepos/a10a20_1/.environment/cipd:/Users/wiederhirn/google-cloud-sdk/bin:/var/folders/pz/ywts5sn50633vt5t3d0wcwvr00tsh6/T/frum_74196_1672857088360/bin:/Users/wiederhirn/.cargo/bin:/Users/wiederhirn/.local/bin:/usr/local/git/git-google/bin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Applications/Utilities/iTerm.app/Contents/Resources/utilities:/Users/wiederhirn/.fig/bin:/Users/wiederhirn/.local/bin:/usr/local/opt/fzf/bin:/Users/wiederhirn/.local/bin:/Users/wiederhirn/.cargo/bin:/usr/local/lib/ruby/gems/3.1.0/bin:/usr/local/opt/fzf/bin:/Users/wiederhirn/Library/Application Support/JetBrains/Toolbox/scripts:/Users/wiederhirn/.fig/bin:/Users/wiederhirn/Library/Application Support/JetBrains/Toolbox/scripts
    - FIG_HOSTNAME: wiederhirn-macbookpro.roam.internal
    - __CFBundleIdentifier: com.googlecode.iterm2
    - FIG_TERM: 2.10.1-beta.0
```

**Comment by brendanfalk** [comment]
Temporary fix: Please add `# Fig Ignore` anywhere in your dotfile (bashrc, zshrc etc) and Fig will ignore the fact that it is not sourced at the top

**Comment by gwiede** [comment]
As of what version?  I've done so immediately below the shebang line in my .bashrc and I'm still seeing the same "Fix hasn't been configured correctly" warnings and offers to fix configuration.  

Further, the instant I saved the .bashrc containing the `# Fig Ignore` line, Fig invisibly disappeared from menu bar (without crash reporter or anything).  When I ran it again, it had the same error messages per above.

**Comment by brendanfalk** [comment]
@grant0417

**Comment by grant0417** [comment]
Can you run `fig doctor --verbose` and share the output @gwiede

**Comment by gwiede** [comment]
Ah, that solved it, I needed it in both .bashrc & .profile, but only added it to .bashrc.  Now appears to be working fine, thanks!!

I presume that also means Fig no longer does error detection/fixup when stuff is added at end any longer?  If so, fine for short- to mid-term, but would still prefer real solution (actual ignoring of comments, but detecting/fixing when active content added to beginning or end) sooner versus later.

**Comment by grant0417** [comment]
This will be fixed in 2.12, so you will be able to remove the fix then, it will actually ignore comments.

**Comment by grant0417** [comment]
The fix should be live on 2.12.0, you can remove the work around now

**Comment by gwiede** [comment]
Thanks muchly!
